contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql

```diff
@@ -31,6 +31,26 @@ end
 $$
 LANGUAGE plpgsql;
 
+CREATE OR REPLACE FUNCTION sys.babelfish_update_server_collation_name() RETURNS VOID
+LANGUAGE C
+AS 'babelfishpg_common', 'babelfish_update_server_collation_name';
+
+SELECT sys.babelfish_update_server_collation_name();
+
+DROP FUNCTION sys.babelfish_update_server_collation_name();
+
+-- reset babelfishpg_tsql.restored_server_collation_name GUC
+do
+language plpgsql
+$$
+    declare
+        query text;
+    begin
+        query := pg_catalog.format('alter database %s reset babelfishpg_tsql.restored_server_collation_name', CURRENT_DATABASE());
+        execute query;
+    end;
+$$;
```

Comment:
I am thinking if we re-set this guc once in the first upgrade script then what happens when we again execute the above procedure in the next upgrade script with the default value (NULL?). Are we handling this in the `C` implementation of the procedure?

Reply:
yes, this will cause an issue as we are raising error if restored_collation_name is NULL

Reply:
But that doesn't sound right isn't it? We are going to repeatedly call this procedure and reset this GUC in each upgrade script so does that mean we are going to fail the second time we call this procedure since the GUC would have been reset now?

Reply:
Updated in latest rev.